### PR TITLE
s3: Add Magalu S3 Object Storage as provider

### PR DESCRIPTION
**Pull request by folkzb**
#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by albertony** [comment]
Thanks. You have some changes in autogenerated parts, which you can remove:
- Everything in `MANUAL.*` files.
- In `s3.md` between:

  `{{< rem autogenerated options start" - DO NOT EDIT - instead edit fs.RegInfo in backend/s3/s3.go then run make backenddocs" >}}`

  and

  `{{< rem autogenerated options stop >}}`

File: backend/s3/s3.go [hunk 1]
```diff
@@ -3173,6 +3206,8 @@ func setQuirks(opt *Options) {
 	case "LyveCloud":
 		useMultipartEtag = false // LyveCloud seems to calculate multipart Etags differently from AWS
 		useAlreadyExists = false // untested
+	case "Magalu":
```
Comment (ncw) [inline backend/s3/s3.go:3209]: Did you run the integration tests to check these are the correct quirks?

**Review by ncw — COMMENTED** [review]
Looking good - well done :-)

See @albertony 's comments on autogenerated info.

Did you check out 

https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#adding-a-new-s3-provider

Thank you

File: backend/s3/s3.go (hunk 1 above)
Comment (ncw) [reply backend/s3/s3.go:3209]: Great - well done.

The Multithread Copy errors are most likely a bug in the service.

Comment (folkzb) [reply backend/s3/s3.go:3209]: In my view, the pr is ready, does it need anything else @ncw ?

**Review by ncw — APPROVED** [review]
This looks great now - thank you :-)

I'll merge it now.

**Pull request merged**